### ISSUE TITLE
Use after free in exceptions

### ISSUE THREAD

**Issue by gicmo** [G-Node/nix#462]
As found by coverity we have server exceptions with the following pattern:

```cpp
 const char *what() const NOEXCEPT {
        std::stringstream sstream("EmptyString: ");
        sstream << "Empty string given - not a valid value for " << caller;
        return sstream.str().c_str();
    }
```

What is going on is that once the function (`what()`) returns, the local auto variable `sstream` gets deallocated and with it the string it contained (`sstream.str()`). The `c_str()` member of that string returns a *pointer to const* of its internal data, which is gone after the string is deallocated. Hence the use-after-free error. 

- [x] 87026 `UnsortedTicks`
- [x] 87027 `MissingAttr`
- [x] 87029 `MisingName`
- [x] 87030 `EmptyString`
- [x] 87032 `FileNotOpen`
- [x] 87039 `IncompatbileDimension`
- [x] 87043 `InvalidUnit`
- [x] 87051 `InvalidDimension`
- [x] 87053 `FileNotFound`
- [x] 87054 `InvalidName`

**Comment by gicmo** [comment]
Should be fixed, but I want to run coverty again before closing them.

**Comment by jgrewe** [comment]
closed via pr #540
